### ISSUE TITLE
Test for AND condition with same variable- var=0 && var=1

### ISSUE THREAD

**Issue by PeterPitterling** [koalaman/shellcheck#2380]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

_rc=1

if [[ ${_rc} -eq 1 && ${_rc} -eq 2 ]]; then

 : # could never be reached as test is never true 

fi
```

#### Here's what shellcheck currently says:
no issue detected

#### Here's what I wanted or expected to see:
test is never true .. did you mean OR ||

**Comment by e-kwsm** [comment]
Now we have [SC2333](https://shellcheck.net/wiki/SC2333).
